hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -305,14 +286,17 @@ class IncrementalRelation(val sqlContext: SQLContext,
   }
 
   private def fullTableScanDataFrame(startInstantTime: String, endInstantTime: String): DataFrame = {
+    val commitTimesToReturn = commitsToReturn.map(_.getTimestamp)
     val hudiDF = sqlContext.read
       .format("hudi_v1")
       .schema(usedSchema)
       .load(basePath.toString)
-      .filter(String.format("%s > '%s'", HoodieRecord.COMMIT_TIME_METADATA_FIELD, //Notice the > in place of >= because we are working with optParam instead of first commit > optParam
+      // using closed_closed range here since we are using start instant from QueryContext
+      .filter(String.format("%s >= '%s'", HoodieRecord.COMMIT_TIME_METADATA_FIELD,
         startInstantTime))
       .filter(String.format("%s <= '%s'", HoodieRecord.COMMIT_TIME_METADATA_FIELD,
```

Comment:
do we still need the `<=` and `>=` predicates because we already have the `in` predicate.

Reply:
I think we can remove it as the `IN` predicate should also enable predicate pushdown and efficient filtering with column ranges in parquet.

Reply:
> IN predicate should also enable predicate pushdown

In predicate is indeed not very friendly for predicates pushdown, let's make it clear for this.